pkg/kubelet/lifecycle/handlers.go

```diff
@@ -99,6 +99,9 @@ func resolvePort(portReference intstr.IntOrString, container *v1.Container) (int
 	}
 	for _, portSpec := range container.Ports {
 		if portSpec.Name == portName {
+			if portSpec.Protocol != v1.ProtocolTCP {
```

Comment:
is name required to be unique or could there be another port with the same name and a TCP protocol?

Reply:
Yes. this must be an unique 

https://github.com/kubernetes/kubernetes/blob/4f9317596ccb9fcb37bf183babf3e8640afcca9b/staging/src/k8s.io/api/core/v1/types.go#L1821-L1827

https://github.com/kubernetes/kubernetes/blob/4f9317596ccb9fcb37bf183babf3e8640afcca9b/pkg/apis/core/validation/validation.go#L2112-L2128

Reply:
a brief comment here would be helpful.  I had a similar thought when reviewing